url2df0f7c811_przemekgalazka_legacycode_tgz-pJ8-pl_cc_core_cmd_CommandJ8/src/ccproxy/GadajAsterisk.java

```diff
@@ -0,0 +1,695 @@
+			asteriskPort = null;
+			numer = 0;
+			asteriskIP = null;
+			asteriskUser = null;
+			asteriskPass = null;
+
+			echoSocket = null;
+			out = null;
+			in = null;
+			asteriskReader = true;
+			connected = true;
+
+		} catch (IOException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+	}
+
+	public void sendAction(String action) {
+		out.print("Action: " + action + "\r\n\r\n");
+		out.flush();
+	}
+
+	private boolean authAsterisk() throws IOException {
+		String line = in.readLine();
+		System.out.println(line);
+
+		out.print("Action: login\r\n");
+		out.print("username: " + asteriskUser + "\r\n");
+		out.print("secret: " + asteriskPass + "\r\n");
+		out.print("\r\n");
+		out.flush();
+		line = in.readLine();
+		System.out.println(line);
+		if (line.equals("Response: Success")) {
+			return true;
+		}
+
+		return false;
+	}
+
+	public void run() {
+		setName("T-GadajAst");
+		try {
+			// sleep(500);
+			sleep(0);
+		} catch (InterruptedException e1) {
+			// TODO Auto-generated catch block
+			e1.printStackTrace();
+		}
+		try {
+		if (asteriskReader) {
+			setName("T-GadajAstReader");
+			MessageFromAsterisk message = new MessageFromAsterisk(p);
+            boolean fullybooted=false;
+            String line="";
+            log.info("Waiting for Asterisk to fully boot...");
+            while (!fullybooted||line.length() != 0){
+                line = in.readLine();
+                if (line.equals("Event: FullyBooted")) fullybooted=true;
+            }
+            log.info("Asterisk fully booted");
+			while (true) {
+				try {
+					line = in.readLine();
+					logMessageFromAsterisk.trace(line);
+					// System.out.println(line);
+					if (line != null) {
+
+						// p.log.info("ok, line: "+ line);
+						if (line == null)
+							continue;
+						if (line.length() == 0) {
+							message.showEvent();
+							message.parseEvent(wiadomosciOdAsteriska);
+							// message.send();
+							message = new MessageFromAsterisk(p);
+							// p.asteriskQueueList.showQueues();
+						
+						} else
+							message.parseLine(line);
+					} else {
+						connected = false;
+						sleep(2000);// zeby nie bylo exceptionsow,bo musi sie
+									// nowe polaczenie zestawić i w ogole
+					}
+
+				} catch (IOException e) {
+					// TODO Auto-generated catch block
+					e.printStackTrace();
+				} catch (InterruptedException e) {
+					// TODO Auto-generated catch block
+					e.printStackTrace();
+				}
+			}
+		} else {
+			while (true) {
+				try {
+					if (!wyslijWiadomosci() & !wyslijOdpowiedz()) 
+						sleep(100);
+					
+				} catch (InterruptedException e) {
+					// TODO Auto-generated catch block
+					e.printStackTrace();
+				}
+			}
+		}
+		} catch (Exception e) {
+			e.printStackTrace();
+			System.exit(-1);
+		}
+	}
+
+	public void sendMessage(AgentConnection agent, String outputLine) {
+		sendMessage(agent, outputLine, true);
+	}
+
+	public void sendMessage(AgentConnection agent, String outputLine,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.